From 796328928c104b0c470109a959beef93374273f6 Mon Sep 17 00:00:00 2001
From: David L Kreitzer <david.l.kreitzer@intel.com>
Date: Wed, 18 Sep 2013 15:21:59 -0700
Subject: Dalvik: Improve the translation of the FP compares in the PCG glue layer

BZ: 139584

Replace the "feq" conditions that we are currently using to distinguish between
0 and 1 results with conditions that can be correctly implemented with a single
branch.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ifb50160a367b7914300a28ea5cd52e3cc46e4eac
Orig-MCG-Change-Id: Ic832c91d3785a0da205830ca54bd9801c4ffe753
Signed-off-by: Kreitzer, David L <david.l.kreitzer@intel.com>
Reviewed-on: http://android.intel.com:8080/132985
Reviewed-by: Lupusoru, Razvan A <razvan.a.lupusoru@intel.com>
Reviewed-by: Hartley, Timothy D <timothy.d.hartley@intel.com>
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Provodin, Vitaly A <vitaly.a.provodin@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/LowerJump.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/LowerJump.cpp b/vm/compiler/codegen/x86/pcg/LowerJump.cpp
index f97cc15..2bb7c7a 100644
--- a/vm/compiler/codegen/x86/pcg/LowerJump.cpp
+++ b/vm/compiler/codegen/x86/pcg/LowerJump.cpp
@@ -155,11 +155,11 @@ void dvmCompilerPcgTranslateIfFp (CompilationUnitPCG *cUnit, MIR *mir, uint32_t
     {
         case 1:
             cond1 = "flt";
-            cond2 = "feq";
+            cond2 = "fle";
             break;
         case -1:
             cond1 = "fnge";
-            cond2 = "feq";
+            cond2 = "fngt";
             break;
         default:
             // Currently, this case is never hit.  I am not sure whether there is
-- 
1.7.4.1

